File: src/main/java/org/dxworks/githubminer/dto/export/BranchDTO.java
package org.dxworks.githubminer.dto.export;

import com.google.api.client.util.Key;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.dxworks.githubminer.dto.response.repository.branches.Branch;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class BranchDTO {
    @Key
    private String name;
    @Key
    private String commit;

    public static BranchDTO fromBranch(Branch branch) {
        if (branch == null || com.google.api.client.util.Data.isNull(branch))
            return null;
        return builder()
                .name(branch.getName())
                .commit(branch.getSha())
                .build();
    }
}

File: src/main/kotlin/org/dxworks/githubminer/dto/export/BranchDTO.kt
package org.dxworks.githubminer.dto.export

import com.google.api.client.util.Data
import com.google.api.client.util.Key
import org.dxworks.githubminer.dto.response.repository.branches.Branch

class BranchDTO {
    @Key
    var name: String? = null

    @Key
    var commit: String? = null

    companion object {
        fun fromBranch(branch: Branch?): BranchDTO? {
            return if (branch == null || Data.isNull(branch)) null else BranchDTO().apply {
                name = branch.name
                commit = branch.sha
            }
        }
    }
}
